Source/CombatExtended/CombatExtended/ArmorUtilityCE.cs

```diff
@@ -115,13 +115,17 @@ public static DamageInfo GetAfterArmorDamage(DamageInfo originalDinfo, Pawn pawn
                 // Apparel is arranged in draw order, we run through reverse to go from Shell -> OnSkin
                 for (int i = apparel.Count - 1; i >= 0; i--)
                 {
-                    if (apparel[i].def.apparel.CoversBodyPart(hitPart) 
-                        && !TryPenetrateArmor(dinfo.Def, apparel[i].GetStatValue(dinfo.Def.armorCategory.armorRatingStat), ref penAmount, ref dmgAmount, apparel[i]))
+                    var app = apparel[i];
+
+                    if (app != null
+                        && app.def.apparel.CoversBodyPart(hitPart)
+                        && !TryPenetrateArmor(dinfo.Def, app.GetStatValue(dinfo.Def.armorCategory.armorRatingStat), ref penAmount, ref dmgAmount, app))
                     {
                         // Hit was deflected, convert damage type
                         //armorReduced = true;
                         dinfo = GetDeflectDamageInfo(dinfo, hitPart);
-                        i++;    // We apply this piece of apparel twice on conversion, this means we can't use deflection on Blunt or else we get an infinite loop of eternal deflection
+                        if (app == apparel.ElementAtOrDefault(i))   //Check whether the "deflecting" apparel is still in the WornApparel - if not, it has been destroyed by the damage

```

Comment:
If the goal is to check for destruction, why not use .Destroyed?

Reply:
What this method does is check that the reference to the apparel (app) is still present at the position apparel[i]. The reference will remain valid, but the list apparel will not contain app anymore.

I have no idea why a piece of apparel is iterated twice in this method, but it is the source of the error as it will check the apparel[i] twice, and if i is equal to the length of WornApparel (e.g outer layer of the pawn is destroyed), the method crashes. This only happens when the type of damage done to the apparel isn't sharp damage (I think?) and a whole bunch of other factors - which comes down to the error only popping up whenever a backpack (which is always the outer layer of the pawn) is destroyed while the method determines a deflection has occurred, then the apparel is checked twice and the method figures it ain't there anymore.

E.g I don't know. Maybe it is destroyed, maybe not. It's just not in the list anymore.

Reply:
Latest commit changes the comment to not mention "destroyed" because I have no idea but it works

Reply:
The repeat iteration happens on deflection: sharp damage is converted to blunt and reapplied to the same piece of armor in order to check against it's blunt rating.